cmd/syncProcessor.go

```diff
@@ -175,9 +176,32 @@ type localFileDeleter struct {
 	rootPath string
 }
 
+// As at version 10.4.0, we intentionally don't delete directories in sync,
+// even if our folder properties option suggests we should.
+// Why? The key difficulties are as follows, and its the third one that we don't currently have a solution for.
+// 1. Timing (solvable in theory with FolderDeletionManager)
+// 2. Identifying which should be removed when source does not have concept of folders (e.g. BLob)
+//    Probably solution is to just respect the folder properties option setting (which we already do in our delete processors)
+// 3. In Azure Files case (and to a lesser extent on local disks) users may have ACLS or other properties
+//    set on the directories, and wish to retain those even tho the directories are empty. (Perhaps less of an issue
+//    when syncing from folder-aware sources that DOES NOT HAVE the directory. But still an issue when syncing from
+//    blob. E.g. we delete a folder because there's nothing in it right now, but really user wanted it there,
+//    and have set up custom ACLs on it for future use.  If we delete, they lose the custom ACL setup.
+// TODO: shall we add folder deletion support at some stage? (In cases where folderPropertiesOption says that folders should be processed)
```

Comment:
This sounds like it goes hand-in-hand with the overwrite option.

Reply:
Yeah, let's address it as part of that. (I.e. not in this PR)